iodata/formats/extxyz.py

```diff
@@ -0,0 +1,192 @@
+                                 (lambda word: -float(word)),
+                                 (lambda value: "{:15.10f}".format(-value)))}
+    atnum_column = ("atnums", None, (), int,
+                    (lambda word: int(word) if word.isdigit() else sym2num[word.title()]),
+                    (lambda atnum: "{:2s}".format(num2sym[atnum])))
+    splitted_properties = properties.split(':')
+    assert len(splitted_properties) % 3 == 0
+    # Each property has 3 values: its name, dtype and shape
+    names = splitted_properties[::3]
+    dtypes = splitted_properties[1::3]
+    shapes = splitted_properties[2::3]
+    if 'Z' in names:
+        # Try to map 'Z' to the 'atnums' attribute
+        atom_column_map['Z'] = atnum_column
+    elif 'species' in names:
+        # If 'Z' is not present, use 'species'
+        atom_column_map['species'] = atnum_column
+    for name, dtype, shape in zip(names, dtypes, shapes):
+        if name in atom_column_map.keys():
+            atom_columns.append(atom_column_map[name])
+        else:
+            # Use the 'extra' attribute to store values which are not predefined in iodata
+            if shape == '1':
+                shape_suffix = ()
+            else:
+                shape_suffix = (int(shape),)
+            atom_columns.append(('extra', name, shape_suffix, *dtype_map[dtype]))
+    return atom_columns
+
+
+def _parse_title(title: str):
+    """Parse the title in an extended xyz file."""
+    key_value_pairs = shlex.split(title)
+    # A dict of predefined iodata atrributes with their names and dtype convertion functions
+
+    def load_cellvecs(word):
+        return np.array(word.split(), dtype=np.float).reshape([3, 3]) * angstrom
+    iodata_attrs = {'energy': ('energy', float),
+                    'Lattice': ('cellvecs', load_cellvecs),
+                    'charge': ('charge', float)}
+    data = {}
+    for key_value_pair in key_value_pairs:
+        if '=' in key_value_pair:
+            key, value = key_value_pair.split('=', 1)
+            if key == 'Properties':
+                atom_columns = _parse_properties(value)
+            elif key in iodata_attrs.keys():
+                data[iodata_attrs[key][0]] = iodata_attrs[key][1](value)
+            else:
+                data.setdefault('extra', {})[key] = _convert_title_value(value)
+        else:
+            # If no value is given, set it True
+            data.setdefault('extra', {})[key_value_pair] = True
+    return atom_columns, data
+
+
+@document_load_one("EXTXYZ", ['title'],
+                   ['atcoords', 'atgradient', 'atmasses', 'atnums', 'cellvecs',
+                    'charge', 'energy', 'extra'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    atom_line = next(lit)
+    title_line = next(lit)
+    # parse title
+    atom_columns, title_data = _parse_title(title_line)
+    lit.back(title_line)
+    lit.back(atom_line)
```

Comment:
Is there anyway to avoid going back on the lines read?

Reply:
I don't think there is an short alternative. Cases like these were the reason for adding the `back` method. This allows `xyz.py` to be re-used without much modification, because it expects the two lines to be present.

Reply:
(I've suggested Maarten to use this trick.)